### PR TITLE
Improvements to Tenant Deletion

### PR DESCRIPTION
**Pull request by esune**
- Improve UI for deleting suspended tenants
- More consistent wording for tenant deletion
- Block API usage for deleted tenants
- Delete API keys when deleting or suspending tenants 

Opening on behalf of @Gavinok

### CODE REVIEW

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py
Comment (loneil) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:542]: So for discussion and any other people's thoughts on what we are doing here (correct me if I'm wrong on any of this)

- We want to stop token creation for "suspended" soft-deleted tenants (IE tenants in the deleted state)
- Token creation is overridden in the other plugin Multitenant Provider
- See that here https://github.com/hyperledger/aries-acapy-plugins/blob/main/multitenant_provider/multitenant_provider/v1_0/routes.py#L76
- We can't put the tenant state check there because in that plugin there isn't any "Tenant" concept, that's Innkeeper plugin and obviously the multitenant provider could be used independently (why use that plugin, see details here https://github.com/hyperledger/aries-acapy-plugins/tree/main/multitenant_provider)

I guess one concern of this here is we are duplicating some code essentially, if some enhancement or fix happens in the other plugin create_token we wouldn't get it here?

There's some other code in the plugin at this line:

https://github.com/hyperledger/aries-acapy-plugins/blob/main/multitenant_provider/multitenant_provider/v1_0/routes.py#L116

where `if (not wallet_record.requires_external_key) and wallet_key:` begins.
That's not replicated here so are we bypassing that logic then? The `always_check_provided_wallet_key` check there... I'm not sure if that's something necessary, need to refresh my memory and test out what it's doing.

Regardless, duplicating this function here in Innkeeper might be necessary for the pattern of preventing tokens for deleted tenants, but if anyone else has any thoughts/opinions can discuss here

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py
Comment (loneil) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:553]: Warn probably should be down to debug

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py
Comment (loneil) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:571]: Would remove this entirely, it's `debug` in the Multitenant Provider, but logging the key in any situation, even debug, probably not good IMO (might want to remove that debug one in the other plugin)

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py
Comment (loneil) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:558]: Warn probably should be down to debug

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py
Comment (loneil) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:1101]: can add a quick code comment here about what/why it's doing

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py
Comment (loneil) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py:112]: Think this is unused?

**Review by loneil — CHANGES_REQUESTED** [review]
Some minor code comments to start, but also want to see if anyone has any thoughts or things to look out for with the pattern of overriding the token route

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py
Comment (esune) [reply plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:542]: Off the top of my head, I think the duplication might be inevitable since we overriding multiple times the same function and there is - in theory - no guarantee that the plugins would be installed and executed in the same order. Which brings me to the follow-up question: would it cause issues to have the plugins installed in reverse order than what it was tested?

i.e.: currently we install multitenant provider and then innkeeper, with multitenant overriding built-in endpoints and innkeeper overriding the result of that. What happens if we reverse the order of installation? If we must install plugins in a specific order we may want to add a note in the docs about that.

Comment (Gavinok) [reply plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:542]: We need to explicitly load multitenant before innkeeper for these overrides to function

Comment (Gavinok) [reply plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py:543]: I have now imported the original function used in the multitenant plugin inside of the innkeeper plugin to reduce the duplication of code
https://github.com/bcgov/traction/blob/block-api-for-deleted-tenant/plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py#L557

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/models.py
Comment (jamshale) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/models.py:364]: I don't mind this explicit for loop but just fyi you could use list comprehension here if you wanted.

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py
Comment (jamshale) [inline plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py:115]: Do you want this to be a print statement?

**Review by jamshale — COMMENTED** [review]
Looking good to me 👍  I'll let the regular maintainers decide to approve.

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/models.py
Comment (Gavinok) [reply plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/models.py:364]: Is there a reason to use list comprehension here if we don't intend on using the list for anything? I may be misunderstanding

Comment (jamshale) [reply plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/models.py:364]: Just if you want to write it in one line. You don't have to use the returned list. Sometimes it's more readable. But I don't know, people have different opinions.

File: plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py
Comment (Gavinok) [reply plugins/traction_innkeeper/traction_innkeeper/v1_0/tenant/routes.py]: Forgot to remove that on this branch. This code is now dropped

**Review by esune — COMMENTED** [review]
Looks good to me, did a quick smoke test and it works as expected. 👍🏻

**Review by loneil — APPROVED** [review]
It's looking good to me, great to have it leverage the multitenant token function (rather than a full override) too

**Pull request merged**
